platform/wayland/cog-utils-wl.c

```diff
@@ -187,6 +187,180 @@ cog_wl_display_destroy(CogWlDisplay *display)
     g_slice_free(CogWlDisplay, display);
 }
 
+static void
+xdg_popup_on_configure(void *data, struct xdg_popup *xdg_popup, int32_t x, int32_t y, int32_t width, int32_t height)
+{
+}
+
+static void
+xdg_popup_on_popup_done(void *data, struct xdg_popup *xdg_popup)
+{
+    cog_wl_popup_destroy((CogWlPopup *) data);
```

Comment:
I am a bit concerned that this will leave a dangling pointer in `platform->popup`, but fixing that would involve a bigger patch and probably turning `CogWlPopup` into a GObject to make `platform->popup` a weak reference that gets reset to `NULL` automatically when the popup is destroyed... Anyway, I am thinking out loud and this is okay for now, no need to change the code.

Reply:
Fixed in https://github.com/Igalia/cog/pull/643